modules/understanding-upgrade-channels.adoc

```diff
@@ -7,159 +7,145 @@
-no impact on the version of the cluster you install; the `openshift-install`
-binary for a given patch level of {product-title} always installs that patch level.
+upgrade channels will never include an upgrade to a 4.5 release. This strategy ensures that
+administrators explicitly decide to upgrade to the next minor version of
+{product-title}. Upgrade channels control only release selection and do not impact the version of the cluster that you install; the `openshift-install`
+binary file for a specific version of {product-title} always installs that version.
 
-{product-title} {product-version} has the following upgrade channels to choose from:
+{product-title} {product-version} offers the following upgrade channels:
 
-* candidate-{product-version}
-* fast-{product-version}
-* stable-{product-version}
+* `candidate-{product-version}`
+* `fast-{product-version}`
+* `stable-{product-version}`
 
 [discrete]
-== candidate-{product-version} Channel
+== candidate-{product-version} channel
 
-The candidate-{product-version} channel will contain candidate builds for a z-stream
+The `candidate-{product-version}` channel contains candidate builds for a z-stream
 ({product-version}.z) release.
-Release candidates contain all the features of the product but they are not supported and
-should only be used to test feature acceptance and assist in qualifying the next version
+Release candidates contain all the features of the product but are not supported. Use release candidate versions to test feature acceptance and assist in qualifying the next version
 of {product-title}.
-A release candidate is any build (e.g. 4.3.0-rc.3, 4.3.0) that is available in the candidate
+A release candidate is any build, such as `4.3.0-rc.3` that is available in the candidate
 channel.
-After a version lands in the candidate channel, it goes through more quality checks and if
-it meets the quality standard it is promoted to fast-{product-version} or stable-{product-version}
-channels.
-So if a given release is available in the candidate-{product-version} channel and also in the fast-{product-version}
-or stable-{product-version} channels, it is a Red Hat supported version.
-Additionally candidate-{product-version} may include dead end releases from which there are no or ever
-be recommended upgrades.
-
-The candidate-{product-version} channel also allows upgrading from a previous minor version of
+After a version is available in the candidate channel, it goes through more quality checks, and if
+it meets the quality standard, it is promoted to the `fast-{product-version}` or `stable-{product-version}` channels.
+Because of this testing strategy, if a specific release is available in both the `candidate-{product-version}` channel and in the `fast-{product-version}`
+or `stable-{product-version}` channels, it is a Red Hat supported version.
+The `candidate-{product-version}` channel can include release versions that you can never upgrade from.
```

Comment:
I think there's a useful distinction between upgrade possibilities, which you _can_ attempt and which might succeed, and recommended updates.  Channels are our recommended updates.  If you actually attempt those updates, you have a good chance of succeeding, but you might fail.  If you attempt an update that we do not recommend, your odds are probably not as good, but there's still a chance that you succeed for at least some version pairs.  How about:

> The `candidate-{product-version}` channel can include release versions from which there are no recommended updates in any channel.

Re: _in any channel", we will eventually end-of-life 4.1.z, after which the last 4.1.z will have no outgoing edges _in stable-4.1_.  But you'll be able to change into stable-4.2 and update out of that 4.1.z tip and on into 4.2 releases.  Candidate channels are special in that we are fine just writing off a broken release and never provide folks with any offramp.

Complicating this a bit further, I have fuzzy recollections that that long-term-support channels will continue to see z-stream releases beyond the point where updates to the next minor version are supported, and folks will be recommended to use some non-cluster-version-operator migration tool?  If that's true, we probably want text talking about it somewhere in this file.

Reply:
Found some internal docs talking about a migration tool like [the 4.3 -> 4.4 tool][1] for things like 4.1 -> 4.3 updates.  But the docs said that we'd support 4.y -> 4.(y+1) updates via the cluster-version operator for the life of 4.y.z, so no need to pull in migrations tooling here; just ignore that last paragraph in my earlier comment.

[1]: https://docs.openshift.com/container-platform/4.2/migration/migrating_3_4/deploying_cam.html